tasks/main.yml

```diff
@@ -189,15 +162,16 @@
     # extract file name that needs to be postmap(8)ed from `postfix_tables`
     postfix_text_databases: "{{ postfix_tables | selectattr('type', 'match', '^(btree|cdb|dbm|hash|lmdb|sdbm)$') | map(attribute='name') | list }}"
 
-- name: Run postmap(1) on changed files by Makefile
-  # set --no-print-directory when make is gmake.
-  #
-  # gmake prints "Entering/Leaving directory ..." even when there is nothing to
-  # do
-  shell: "make -C {{ postfix_db_dir }} {% if postfix_make_is_gmake %}--no-print-directory {% endif %}HASHES='{{ postfix_text_databases | join(' ')  }}'"
-  register: register_make_postmap
-  changed_when: register_make_postmap.stdout | length > 0
-  notify: Reload postfix
+- name: Run postmap(1) on changed text databases
+  command: "postmap {{ item.item.type }}:{{ item.invocation.module_args.dest }}"
+  with_items: "{{ register_postfix_create_lookup_table.results }}"
+  # translation: run postmap when the table is one of text-based database, and
+  # the table has changed or its database file does not exist
+  when:
+    - item.item.name in postfix_text_databases
+    # for a reason I do not know, "not (var + 'string' | exists)" does not work
+    # here
+    - "item.changed or not [item.invocation.module_args.dest, 'db'] | join('.') | exists"
```

Comment:
"item.changed or not ( [item.invocation.module_args.dest, 'db'] | join('.') | exists )"
using parentheses in a long expression helps to read and reduces mistakes.